gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/infra/crud_service/portal_page/PortalPageCrudServiceImpl.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright © 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.apim.infra.crud_service.portal_page;
+
+import io.gravitee.apim.core.portal_page.crud_service.PortalPageCrudService;
+import io.gravitee.apim.core.portal_page.model.PortalPage;
+import io.gravitee.apim.core.portal_page.model.PortalViewContext;
+import io.gravitee.apim.infra.adapter.PortalPageAdapter;
+import io.gravitee.repository.exceptions.TechnicalException;
+import io.gravitee.repository.management.api.PortalPageContextRepository;
+import io.gravitee.repository.management.api.PortalPageRepository;
+import java.util.List;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+public class PortalPageCrudServiceImpl implements PortalPageCrudService {
+
+    @Lazy
+    private final PortalPageRepository portalPageRepository;
+
+    @Lazy
+    private final PortalPageContextRepository portalPageContextRepository;
+
+    private final PortalPageAdapter portalPageAdapter;
+
+    @Override
+    public List<PortalPage> byPortalViewContext(String environmentId, PortalViewContext portalViewContext) {
```

Comment:
I think it should be the responsability of a domain service to fetch the contexts with its page contents.
 
In the futur using the PortalPage and PortalPageContext crud services should only allow us to fetch the contexts or the pages contents but not both in a single crud service.

Using a dedicated context object allow us to give a light view of available pages to the portal. Content should only be fetched when necessary.